src/visualization/interaction/Highlighter.js

```diff
@@ -35,62 +36,43 @@ ROS3D.Highlighter.prototype.onMouseOver = function(event) {
  * @param event - the event that contains the target of the mouseout
  */
 ROS3D.Highlighter.prototype.onMouseOut = function(event) {
-  this.hoverObjs.splice(this.hoverObjs.indexOf(event.currentTarget), 1);
+  // this.hoverObjs.splice(this.hoverObjs.indexOf(event.currentTarget), 1);
+  this.highlightObject(event.currentTarget, false);
 };
 
+
 /**
- * Add all corresponding webgl objects in the given scene and add them to the given render list.
+ * Highlight and unhighlight the given object
  *
- * @param scene - the scene to check for webgl objects
- * @param objects - the objects list to check
- * @param renderList - the list to add to
+ * @param object - the target object to (un)highlight
+ * @param flag - whether to highlight or unhighlight
  */
-ROS3D.Highlighter.prototype.getWebglObjects = function(scene, objects, renderList) {
-  var objlist = scene.__webglObjects;
-  // get corresponding webgl objects
-  for ( var c = 0; c < objects.length; c++) {
-    if (objects[c]) {
-      for ( var o = objlist.length - 1; o >= 0; o--) {
-        if (objlist[o].object === objects[c]) {
-          renderList.push(objlist[o]);
-          break;
-        }
+ROS3D.Highlighter.prototype.highlightObject = function (object, flag) {
```

Comment:
This doesn't actually highlight correctly, but instead replaces the highlighted objects' materials with a half-transparent white material:
<img width="802" alt="screen shot 2017-12-11 at 15 04 47" src="https://user-images.githubusercontent.com/871405/33834845-c00345b4-de84-11e7-8c30-cc49ee02d074.png">

The old implementation rendered that material *over* the already-rendered scene to achieve the highlighted look. 

Looks like current Three.js has a couple ways to make highlighting happen:
[This demo](https://threejs.org/examples/?q=out#webgl_postprocessing_outline) ([source](https://github.com/mrdoob/three.js/blob/master/examples/webgl_postprocessing_outline.html)) uses an ```OutlinePass``` to draw outlines, which looks really nice but is slightly more involved ([here](https://stackoverflow.com/questions/26341396/outline-a-3d-object-in-three-js)'s a StackOverflow question with some very concise example code).

[This one](https://threejs.org/examples/#webgl_interactive_cubes) ([source](https://github.com/mrdoob/three.js/blob/master/examples/webgl_interactive_cubes.html)) simply gives the material a bit of (red) ```emissive``` lighting to achieve about the same effect we were using before (note they use ```currentHex``` to save the old value – I'd prefer to use ```originalHex```).

Reply:
The second example is the one I referenced to update the highlighting. Thanks for the review. I will take a look again.